core/chains/evm/logpoller/log_poller.go

```diff
@@ -217,6 +218,12 @@ func (filter *Filter) Contains(other *Filter) bool {
 	if other == nil {
 		return true
 	}
+	if other.Retention != filter.Retention {
+		return false
+	}
+	if other.MaxLogsKept != filter.MaxLogsKept {
+		return false
+	}
```

Comment:
I don't think there are any products who actual use the "update a filter" feature (same filter name with different addresses or events), but for consistency we should have added the `other.Retention` check when that was added. Adding this now so that `Retention` and `MaxlogsKept` can be updated dynamically if needed.